src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts

```diff
@@ -39,82 +41,69 @@ export class ViewConfigItemComponent implements OnInit, OnChanges {
         configAttributes = sortBy(configAttributes, ['order', 'description']);
         changes.categoryConfigurationData.currentValue.value = configAttributes;
         this.categoryConfiguration = changes.categoryConfigurationData.currentValue;
+        configAttributes.forEach(el => {
+          this.configItems.push({
+            [el.key]: el.value,
+            type: el.type
+          });
+        });
       }
     }
   }
 
-  public restoreConfigFieldValue(configItemKey, categoryKey, configValue, configType) {
-    let itemKey = categoryKey.toLowerCase() + '-' + configItemKey.toLowerCase();
-    if (itemKey.includes('select')) {
-      itemKey = itemKey.replace('select-', '');
-    }
-    this.isEmptyValue = false;
-    let htmlElement: any;
-    htmlElement = <HTMLInputElement>document.getElementById(itemKey);
-    if (htmlElement == null) {
-      htmlElement = <HTMLSelectElement>document.getElementById('select-' + itemKey);
-      htmlElement.selectedIndex = (this.selectedValue === 'true' ? 1 : 0);
-    } else {
-      htmlElement.value = htmlElement.textContent;
-    }
-
-    const cancelButton = <HTMLButtonElement>document.getElementById('btn-cancel-' + itemKey);
-    cancelButton.classList.add('hidden');
-
-    if (configType.toUpperCase() === 'JSON') {
-      this.isValidJson = ConfigTypeValidation.isValidJsonString(configValue);
-      return;
+  public difference(obj, bs) {
+    function changes(object, base) {
+      return transform(object, function (result, value, key) {
+        if (!isEqual(value, base[key])) {
+          result[key] = (isObject(value) && isObject(base[key])) ? changes(value, base[key]) : value;
+        }
+      });
     }
+    return changes(obj, bs);
   }
 
-  public saveConfigValue(categoryName, configItem, type) {
-    const catItemId = categoryName.toLowerCase() + '-' + configItem.toLowerCase();
-    this.selectedCategoryId = catItemId;
-    let htmlElement: any;
-    htmlElement = <HTMLInputElement>document.getElementById(catItemId);
-
-    let value;
-    if (htmlElement == null) {
-      htmlElement = <HTMLSelectElement>document.getElementById('select-' + catItemId);
-      value = htmlElement.options[htmlElement.selectedIndex].value;
-      this.isEmptyValue = false;
-    } else {
-      value = htmlElement.value.trim();
-      if (value.length === 0) {
-        this.isEmptyValue = true;
-        return;
+  public saveConfiguration(form: NgForm) {
+    const updatedRecord = [];
+    const formData = form.value;
+    for (const key in formData) {
+      if (formData.hasOwnProperty(key)) {
+        updatedRecord.push({
+          [key]: formData[key]
+        });
       }
     }
+    const diff = this.difference(updatedRecord, this.configItems);
+    this.configItems.forEach(item => {
+      for (const key in item) {
+        diff.forEach(changedItem => {
+          for (const k in changedItem) {
+            if (key === k && item[key] !== changedItem[k]) {
+              item[key] = changedItem[k];
```

Comment:
what if save call fails, then you retain the latest changed into this.configItems?